app/hooks/useModeratorRole.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState, useEffect } from 'react';
+
+import { getCookie } from '../utils/cookie';
+
+import { hasGlobalModeratorRole, hasCommunityModeratorRole } from '../utils/properties';
+
+import history from '../createdHistory';
+
+export const useModeratorRole = (communityId, redirectPage) => {
```

Comment:
these hooks are almost the same, combine them, example
```
export const useModeratorRole = (redirectPage, communityId = null) => {

setModeratorRole(
            hasGlobalModeratorRole(permissions) || 
            communityId && hasCommunityModeratorRole(permissions, parseInt(communityId))
);
```

Reply:
+